pkg/types/defaults/installconfig.go

```diff
@@ -80,6 +81,9 @@ func SetInstallConfigDefaults(c *types.InstallConfig) {
 		awsdefaults.SetPlatformDefaults(c.Platform.AWS)
 	case c.Platform.Azure != nil:
 		azuredefaults.SetPlatformDefaults(c.Platform.Azure)
+		if c.Platform.Azure.CloudName == azure.StackCloud {
```

Comment:
this should also check if creds are already set

Reply:
Is there a reason not to put this inside `azuredefaults.SetPlatformDefaults`?